### PR TITLE
feat(HMS-1210): add launch template to wizard

### PR DESCRIPTION
**Pull request by amirfefer**
Screen Shot 2023-03-19 at 19 04 10

### CODE REVIEW

File: src/Components/ProvisioningWizard/steps/AccountCustomizations/aws.js [hunk 1]
```diff
@@ -67,6 +68,37 @@ const AccountCustomizationsAWS = ({ setStepValidated, architecture, composeID })
...
         <RegionsSelect composeID={composeID} />
       </FormGroup>
+      <FormGroup
+        label="Select template"
+        fieldId="aws-select-template"
+        labelIcon={
+          <Popover
+            bodyContent={
+              <span>
+                Launch templates contains the configuration information to launch an instance, For further information and for creating launch
```
Comment (amirfefer) [inline src/Components/ProvisioningWizard/steps/AccountCustomizations/aws.js:78]: @lzap, here is the help text for it, I also added a link for more details

Comment (lzap) [reply src/Components/ProvisioningWizard/steps/AccountCustomizations/aws.js]: Great thanks.

File: src/Components/ProvisioningWizard/steps/ReservationProgress/index.js [hunk 2]
```diff
@@ -52,21 +53,24 @@ const ReservationProgress = ({ setLaunchSuccess }) => {
...
   });
 
+  const reservationParams = () => ({
+    source_id: chosenSource,
+    [instanceType(provider)]: chosenInstanceType,
+    amount: chosenNumOfInstances,
+    image_id: chosenImageID,
+    [region(provider)]: chosenRegion,
+    pubkey_id: chosenSshKeyId,
+    launch_template_id: chosenTemplate,
```
Comment (ezr-ondrej) [inline src/Components/ProvisioningWizard/steps/ReservationProgress/index.js:63]: This is just for AWS, I think the parameter would be safely ignored for other providers, but I wonder if we want to address this by having params function that would accept provider as parameter?

**Comment by ezr-ondrej** [comment]
As discussed offline, we should have a feature flag for this feature.

**Comment by akhil-jha** [comment]
I think we need a rebase here :)

**Comment by amirfefer** [comment]
Rebased and also this feature is now  gated by the feature flag `provisioning.aws.templates`

**Review by akhil-jha — APPROVED** [review]

**Pull request merged**